### PR TITLE
feat: add basic plugin template framework

### PR DESCRIPTION
**Pull request by leslie-tsang**
This PR add basic plugin template framework for `apisix-plugin-template`

## framework's workflow
1. clone `apisix` repo to `workbench` folder
2. inject custom plugin and test cases to original apisix, which will overwrite the original one
3. setup CI env
4. run test case

## TODO

- [x] add basic plugin template
- [x] add test case
- [x] add CI support

### CODE REVIEW

File: t/demo/demo.t [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+=== TEST 1: hello, world
+This is just a simple demonstration of the
+echo directive provided by ngx_http_echo_module.
+--- config
+    location = /t {
+        content_by_lua_block {
+            ngx.say('hello, world!')
+        }
+    }
+--- request
```
Comment (spacewander) [inline t/demo/demo.t]: Let's remove the duplicate sections from the beginning with https://github.com/apache/apisix/blob/7f9192576702eb3e5ab4764319756e9d91ab83c9/t/plugin/gzip.t#L33

File: t/demo/demo.t [hunk 2]
```diff
@@ -0,0 +1,23 @@
+use t::APISIX 'no_plan';
+
+log_level('debug');
```
Comment (spacewander) [inline t/demo/demo.t]: Better not use `debug` by default, it's too verbose to check

File: t/demo/demo.t (hunk 1 above)
Comment (leslie-tsang) [reply t/demo/demo.t]: Thanks for point it out. fixed.

File: t/demo/demo.t (hunk 2 above)
Comment (leslie-tsang) [reply t/demo/demo.t]: Done

**Review by spacewander — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -21,10 +21,40 @@ jobs:
...
+          sudo make ci-env-up project_compose_ci=ci/pod/docker-compose.common.yml
+        working-directory: workbench
+
+#      - name: Launch APISIX deps env
+#        run: |
+#          sudo make ci-env-up
+#          ./ci/linux-ci-init-service.sh
+#        working-directory: workbench
+
+      - name: Run centos7 docker and mapping apisix into container
```
Comment (membphis) [inline .github/workflows/ci.yml]: I think we should run the test cases in Ubuntu.

Comment (leslie-tsang) [reply .github/workflows/ci.yml]: OK. I will fix it.

Comment (leslie-tsang) [reply .github/workflows/ci.yml]: Done

**Pull request merged**